### PR TITLE
Taproot: Fix p2tr support in `ComputeInputScript`

### PR DESCRIPTION
**Pull request by guggero**
## Change Description
Fixes https://github.com/lightningnetwork/lnd/issues/6678 by correctly using the `PrevOutputs` parameter.

### CODE REVIEW

**Comment by Saitoshi9999** [comment]
🍀

**Review by benthecarman — APPROVED** [review]
tACK, this resolved my bug!

File: lnrpc/signrpc/signer_server.go [hunk 1]
```diff
@@ -519,7 +520,33 @@ func (s *Server) ComputeInputScript(ctx context.Context,
...
 
-	sigHashCache := input.NewTxSigHashesV0Only(&txToSign)
+	var (
+		sigHashCache      = input.NewTxSigHashesV0Only(&txToSign)
+		prevOutputFetcher = txscript.NewMultiPrevOutFetcher(nil)
+	)
+
+	// If we're spending one or more SegWit v1 (Taproot) inputs, then we
+	// need the full UTXO information available.
+	if len(in.PrevOutputs) > 0 {
```
Comment (Roasbeef) [inline lnrpc/signrpc/signer_server.go:530]: Should we also add a check above to error out if this field is set and the sign type _isn't_ taproot?

File: lntest/itest/lnd_taproot_test.go [hunk 2]
```diff
@@ -1043,20 +1043,20 @@ func newAddrWithScript(ctx context.Context, t *testing.T,
...
+		newAddrResp.Address, harnessNetParams,
 	)
 	require.NoError(t, err)
 
-	p2wkhPkScript, err := txscript.PayToAddrScript(p2wkhAddr)
+	pkScript, err := txscript.PayToAddrScript(addr)
 	require.NoError(t, err)
 
-	return p2wkhAddr, p2wkhPkScript
+	return addr, pkScript
```
Comment (Roasbeef) [inline lntest/itest/lnd_taproot_test.go:1162]: 👍

File: lntest/itest/lnd_taproot_test.go [hunk 3]
```diff
@@ -117,6 +118,108 @@ func testTaprootSendCoinsKeySpendBip86(ctxt context.Context,
...
+	}}
+	signResp, err := alice.SignerClient.ComputeInputScript(
+		ctxt, &signrpc.SignReq{
+			RawTxBytes: buf.Bytes(),
+			SignDescs: []*signrpc.SignDescriptor{{
+				Output:     utxoInfo[0],
+				InputIndex: 0,
+				Sighash:    uint32(txscript.SigHashDefault),
+			}},
+			PrevOutputs: utxoInfo,
```
Comment (Roasbeef) [inline lntest/itest/lnd_taproot_test.go:201]: Scratch my prior comment: we detect things based on the type of UTXOs specified and the presence of this field.

**Review by Roasbeef — APPROVED** [review]
LGTM 🐚

**Review by bhandras — APPROVED** [review]
LGTM :+1:

**Pull request merged**